libcontainer/container_linux.go

```diff
@@ -960,12 +974,15 @@ func (c *linuxContainer) criuSwrk(process *Process, req *criurpc.CriuReq, opts *
 
 	// cmd.Wait() waits cmd.goroutines which are used for proxying file descriptors.
 	// Here we want to wait only the CRIU process.
-	st, err := cmd.Process.Wait()
-	if err != nil {
-		return err
-	}
-	if !st.Success() {
-		return fmt.Errorf("criu failed: %s\nlog file: %s", st.String(), logPath)
+	reqType := req.GetType()
+	if reqType != criurpc.CriuReqType_PRE_DUMP {
+		st, err := cmd.Process.Wait()
```

Comment:
checkpoint --pre-dump would block here waiting for something, I don't know the exact reason.
maybe need a  deep dig. for now do not call wait() here when pre-dump works fine for me.